portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,10 +34,55 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private String name;
+  ArrayList<String> comments = new ArrayList<String>();
+  List<Task> tasks = new ArrayList<>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = "Bless";
+    String json = convertToJson();
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Task task = new Task(id, title, timestamp);
+      tasks.add(task);
+    }
+  }
+  private String convertToJson() {
+    Gson gson = new Gson();
+    String json = gson.toJson(tasks);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println(name);
+    response.sendRedirect("/?authuser=0#comments");
+    String title = request.getParameter("title");
```

Comment:
I don't see a `title` name in the form, so this is probably not working. `text-input` is the name you have defined, I believe.